pkg/security/serializers/serializers_linux.go

```diff
@@ -890,6 +892,13 @@ func newProcessContextSerializer(pc *model.ProcessContext, e *model.Event) *Proc
 		ptr = it.Next()
 	}
 
+	// shrink the middle of the ancestors list if it is too long
+	if len(ps.Ancestors) > processTreeMaxDepth {
+		subLen := processTreeMaxDepth / 2
+		ps.Ancestors = append(ps.Ancestors[0:subLen], ps.Ancestors[len(ps.Ancestors)-subLen-processTreeMaxDepth%2:]...)
```

Comment:
I'm not sure to understand the last part `-processTreeMaxDepth%2`, `len(ps.Ancestors)-subLen` isn't enough?

Reply:
`-processTreeMaxDepth%2` is only here to handle the case where `processTreeMaxDepth` is an odd number, this is to make sure the depth of the resulting process tree ends up being equal to `processTreeMaxDepth`.
I'll update the PR to add a comment reflecting this :+1:

Reply:
Got it, thank you ! :bow: